src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -49,6 +49,13 @@ public abstract class NuGetPowerShellBaseCommand : PSCmdlet, IPSNuGetProjectCont
         // TODO: Hook up DownloadResource.Progress event
         private readonly IHttpClientEvents _httpClientEvents;
 
+        // default stopWatch to calculate total time elapsed for an action
+        protected Stopwatch ActionStopWatch = new Stopwatch();
```

Comment:
No, because at many times, we go back and forth from command specific class to base class and have user confirmation dialogs. Since we want to only calculate actual time taken by nuget to process an action and avoiding noise like time to react to a ui dialog, etc... so we need this at this level, otherwise it will be very complicated to add/subtract time duration.